sonar-ruby-plugin/src/main/resources/whitequark_parser_init.rb

```diff
@@ -47,8 +47,9 @@ def initialize(visitor)
 
   def process(node)
     return if node.nil?
+    @visitor.beforeVisit(AstNode.new(node))
     node = super
-    @visitor.visitNode(AstNode.new(node), node.to_a)
+    @visitor.afterVisit(AstNode.new(node), node.to_a)
```

Comment:
See my other comment about calling visit methods individually.
But I think I'd prefer to have something like 
```
tree = @visitor.visitNode(AstNode.new(node), node.to_a)
@visitor.afterVisit(tree)
tree
```